Adds support for batching requests to S3

tomlagier opened the pull request:
## What Changed & Why
This is primarily useful to avoid flooding fake-s3 with network requests.

1) Added "batchSize" configuration option, defaulted to 0
2) Added code path to s3._putObjects to handle batching requests if batchSize option is set
3) Added appropriate tests

## Related issues
See #78 

## PR Checklist
- [x] Add tests
- [x] Add documentation
- [x] Prefix documentation-only commits with [DOC]

## People
@lukemelia

lukemelia commented:
Looks good to me. Can you take a look as well @achambers?

Before we merge, it would great if you can clean up the commit history @tomlagier.

tomlagier committed:
Adds support for batching requests to S3

This is primarily useful to avoid flooding fake-s3 with network requests.

1) Added "batchSize" configuration option, defaulted to 0
2) Added code path to s3._putObjects to handle batching requests if batchSize option is set
3) Added appropriate tests
4) Updated the readme

tomlagier commented:
Squashed the commits, sorry about that.
